### PR TITLE
[Feat] Added dissolving transformation & updated docs

### PR DESCRIPTION
**Pull request by shijianjian**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by shijianjian** [comment]
@johnnv1 I think we may interact more with external libs like diffusers, transformers, etc. Should we add them to our CI for testing? Or any other resolutions?

**Comment by johnnv1** [comment]
> @johnnv1 I think we may interact more with external libs like diffusers, transformers, etc. Should we add them to our CI for testing? Or any other resolutions?

if we go with this external module, i think we can have a specific CI job for testing all cases which depends on these external library... like we have for tutorials, but using or a pytest mark or putting it under some specific module or name on the test suite to select it

**Comment by shijianjian** [comment]
One additional question, like for diffusers, if we need to download the weights for a specific version of diffusion models. It may heavily reduce our CI usability, right? In terms of downloading time, spaces, etc. I am wondering if what is the best approach to test those things.

**Comment by johnnv1** [comment]
> One additional question, like for diffusers, if we need to download the weights for a specific version of diffusion models. It may heavily reduce our CI usability, right? In terms of downloading time, spaces, etc. I am wondering if what is the best approach to test those things.

yeap, you are right. But right now we have already been doing it. We have a bunch of tests that need to download the weights to run. As a palliative measure, I have added a `slow` marker into these tests, so we don't run them against all versions we support.

Ideally, we should have a pre-test job, that downloads all these weights beforehand, and then makes them available before the tests start. This way we can "share" the weights between the actions jobs, and they will not download it when the suite runs

File: docs/source/get-started/highlights.rst [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
 
+At Kornia, we are dedicated to pushing the boundaries of computer vision by providing a robust, efficient, and versatile toolkit. Our library is built on the powerful PyTorch backend, leveraging its efficiency and auto-differentiation capabilities to deliver high-performance solutions for a wide range of vision tasks.
+
+
+Accessible AI Models
+--------------------
+
+We are excited to announce our latest advancement: a new initiative designed to seamlessly integrate lightweight AI models into Kornia. This addition is crafted to empower developers, researchers, and enthusiasts to harness the full potential of accessible AI, simplifying complex vision tasks and accelerating innovation.
+
+We have curated a selection of lightweight AI models, including YuNet, Loftr, and SAM, optimized for performance and efficiency. These models offer efficient computations that do not require expensive GPUs, making cutting-edge AI accessible to everyone. We welcome the whole community of developers and researchers who are passionate about advancing computer vision, throwing PRs for your lightning fast models!
```
Comment (edgarriba) [inline docs/source/get-started/highlights.rst:12]: we might make a call to help curating / updateing more models (SAM2) and to export to onnx

File: kornia/core/external.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import importlib
+
+
+class LazyLoader:
```
Comment (edgarriba) [inline kornia/core/external.py:4]: add some docs anf typing (if makes sense)

File: kornia/core/external.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    def __getattr__(self, item):
+        self._load()
+        return getattr(self.module, item)
+
+    def __dir__(self):
+        self._load()
+        return dir(self.module)
+
+
+diffusers = LazyLoader("diffusers")
```
Comment (edgarriba) [inline kornia/core/external.py:28]: is it needed to register this lib ? or can be done directly where is reallty used, e.g in StableDiffusionDissolving

File: kornia/filters/dissolving.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+          - figure:: https://raw.githubusercontent.com/kornia/data/main/dslv-sd-2.1.png
+
+    Args:
+        version: the version of the stable diffusion model.
+        **kwargs: additional arguments for `.from_pretrained`.
+    """
+
+    def __init__(self, version: str = "2.1", **kwargs):
+        super().__init__()
+        from kornia.core.external import diffusers
```
Comment (edgarriba) [inline kornia/filters/dissolving.py:106]: i would create here directly the lazy loader

**Review by edgarriba — APPROVED** [review]
l like the LazyLoader feature

File: kornia/core/external.py (hunk 3 above)
Comment (shijianjian) [reply kornia/core/external.py:28]: It is better to register here I think. Otherwise, we may have multiple singleton class towards the same lib.

**Comment by shijianjian** [comment]
@johnnv1 Do you have any ideas about how to setup a shared pre-download space? So that I can try adding some tests.

File: kornia/filters/dissolving.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            - torch.sqrt(1.0 / self.scheduler.alphas_cumprod[timestep] - 1) * noise_pred
+        )
+
+    @torch.no_grad()
+    def init_prompt(self, prompt: str) -> None:
+        uncond_input = self.model.tokenizer(
+            [""], padding="max_length", max_length=self.model.tokenizer.model_max_length, return_tensors="pt"
+        )
+        uncond_embeddings = self.model.text_encoder(uncond_input.input_ids.to(self.model.device))[0]
+        text_input = self.model.tokenizer(
```
[the remark this answers is not in the document]
Comment (shijianjian) [reply kornia/filters/dissolving.py:29]: We can. But the problem is that it is too slow. Unless we have some very fast methods. Otherwise, it is not clever to run it in the runtime

File: docs/source/get-started/highlights.rst (hunk 1 above)
Comment (shijianjian) [reply docs/source/get-started/highlights.rst:12]: Probably update the README file to call for model contributors?

**Comment by johnnv1** [comment]
> @johnnv1 Do you have any ideas about how to setup a shared pre-download space? So that I can try adding some tests.

something like

Create a GHA job that will do some torch.load for each weight (need to find out all weights hardcoded) to a specific directory, then upload it to a gha cache (https://github.com/actions/cache), then add download weights step (into the .cache directory) on each test job, then execute the test suite

File: README.md [hunk 6]
```diff
@@ -100,16 +141,20 @@ augmentation_pipeline = AugmentationSequential(
     RandomAffine((-45., 45.), p=1.),
     RandomBrightness((0.,1.), p=1.)
 )
+# Leveraging StableDiffusion models
```
Comment (edgarriba) [inline README.md:130]: can we show above loading images with kornia_rs ?

also I'm not very fond of promoting augmentations as everyone thinks we are only an augmentations library in the end

File: README.md [hunk 7]
```diff
@@ -100,16 +141,20 @@ augmentation_pipeline = AugmentationSequential(
...
 img = augmentation_pipeline(img)
+dslv_op(img, step_number=500)
 
-# Leveraging StableDiffusion models
-output = StableDiffusionDissolving()(img, step_number=500)
-output.save("Kornia-enhanced.jpg")
+dslv_op.save("Kornia-enhanced.jpg")
 ```
 
+<<<<<<< HEAD
```
Comment (edgarriba) [inline README.md]: bad diff

File: README.md (hunk 6 above)
Comment (shijianjian) [reply README.md:130]: Can you modify it directly there? To add more examples or anything else. What I wanted to say here is that we support many different input types and the easy save and visualization.

I think this Quick Start is to promote the use cases in practic. We can include `kornia_rs` as a torch tensor example maybe.

File: README.md [hunk 8]
```diff
@@ -24,11 +24,53 @@ English | [简体中文](README_zh-CN.md)
 </p>
 </div>
 
-**Kornia** is a differentiable computer vision library for [PyTorch](https://pytorch.org).
+**Kornia** is a differentiable computer vision library that provides a rich set of differentiable image processing and geometric vision algorithms. Built on top of [PyTorch](https://pytorch.org), Kornia integrates seamlessly into existing AI workflows, allowing you to leverage powerful [batch transformations](), [auto-differentiation]() and [GPU acceleration](). Whether you’re working on image transformations, augmentations, or AI-driven image processing, Kornia equips you with the tools you need to bring your ideas to life.
```
Comment (edgarriba) [inline README.md:27]: here we are missing some links

File: README.md (hunk 6 above)
Comment (edgarriba) [reply README.md:138]: added

Comment (shijianjian) [reply README.md:138]: We may add a kornia.model example here once the module is ready.

File: README.md [hunk 9]
```diff
@@ -24,11 +24,53 @@ English | [简体中文](README_zh-CN.md)
...
+3. **AI Models**<br>
+Leverage pre-trained AI models optimized for a variety of vision tasks, all within the Kornia ecosystem.
+    - **Face Detection**: YuNet
+    - **Feature Matching**: LoFTR, LightGlue
+    - **Feature Descriptor**: DISK, DeDoDe, SOLD2
+    - **Segmentation**: SAM
+    - **Classification**: MobileViT, VisionTransformer.
 
-It consists of a set of routines and differentiable modules to solve generic computer vision problems. At its core, the package uses *PyTorch* as its main backend both for efficiency and to take advantage of the reverse-mode auto-differentiation to define and compute the gradient of complex functions.
+<details>
```
Comment (shijianjian) [inline README.md:49]: For the best, there should be some code snippet or tutorials for each op.

Comment (edgarriba) [reply README.md:49]: agree -- add as ticket

File: README.md (hunk 6 above)
Comment (edgarriba) [reply README.md:138]: yep, I vote for RTDETR @cjpurackal
